distributed/scheduler.py

```diff
@@ -3854,8 +3853,11 @@ def update_graph(
 
             for a, kv in annotations.items():
                 for k, v in kv.items():
-                    ts = parent._tasks[k]
-                    ts._annotations[a] = v
+                    # Tasks might have been culled, in which case
+                    # we have nothing to annotate.
+                    if k in parent._tasks:
```

Comment:
@jakirkham I haven't followed the performance stuff too closely. Are we to the point where this `if` statement needs profiling?

Reply:
Answered a separate thread ( https://github.com/dask/distributed/pull/4544#pullrequestreview-597956717 ) since it was easier to suggest code that way

We are not asking anyone to do profiling though

Reply:
Thanks!